### PR TITLE
Perf updates for ProxyElements, by returning their content directly

### PR DESCRIPTION
**Pull request by kyleve**
This PR pulls in a reduced subset of #200, which improves performance of deep element hierarchies which use proxy elements. Generally this should save somewhere from 20-40% depending on debug v release mode:

Note: This also includes a delegate to read timing values; can remove.

Debug:
```
Before
---------

Iterations: 48, Average Time: 0.2101164162158966
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_deep_element_hierarchy]' passed (10.143 seconds).

Iterations: 55, Average Time: 0.18392309058796275
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_repeated_layouts]' passed (10.289 seconds).


ProxyElement Changes
---------------------

Iterations: 66, Average Time: 0.15314857526258988
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_deep_element_hierarchy]' passed (10.164 seconds).

Iterations: 61, Average Time: 0.1642437391593808
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_repeated_layouts]' passed (10.209 seconds).
```

Release:
```
Before
---------

Iterations: 113, Average Time: 0.0890231501739637
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_deep_element_hierarchy]' passed (10.128 seconds).

Iterations: 78, Average Time: 0.12875753793960962
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_repeated_layouts]' passed (10.292 seconds).


ProxyElement Changes
---------------------

Iterations: 175, Average Time: 0.05717774868011474
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_deep_element_hierarchy]' passed (10.084 seconds).

Iterations: 90, Average Time: 0.11165261136160956
Test Case '-[BlueprintUICommonControls_Unit_Tests.PerformanceTests test_repeated_layouts]' passed (10.332 seconds).
```

### CODE REVIEW

File: BlueprintUI/Sources/Blueprint View/BlueprintView.swift [hunk 1]
```diff
@@ -48,6 +48,8 @@ public final class BlueprintView: UIView {
             invalidateIntrinsicContentSize()
         }
     }
+    
+    public weak var delegate : BlueprintViewDelegate? = nil
```
Comment (kyleve) [inline BlueprintUI/Sources/Blueprint View/BlueprintView.swift:52]: delegate? or metrics?

Comment (livings124) [reply BlueprintUI/Sources/Blueprint View/BlueprintView.swift:52]: metricsDelegate?

Comment (kyleve) [reply BlueprintUI/Sources/Blueprint View/BlueprintView.swift:52]: I like it!

**Review by kylebshr — APPROVED** [review]
No strong opinion on the timing stuff staying in, seems like something we need a lot

File: BlueprintUICommonControls/Sources/AccessibilityBlocker.swift [hunk 2]
```diff
@@ -19,7 +19,7 @@ public struct AccessibilityBlocker: Element {
     //
 
     public var content: ElementContent {
-        return ElementContent(child: wrapped)
+        wrapped.content
```
Comment (kyleve) [inline BlueprintUICommonControls/Sources/AccessibilityBlocker.swift:22]: Need to double check: Does this work as expected with a custom backingViewDescription?

**Comment by watt** [comment]
> No strong opinion on the timing stuff staying in, seems like something we need a lot

Signpost logging is surely the right answer for this? Are we OK pulling it out once we have that?

**Comment by kyleve** [comment]
Yeah true, though honestly, having the logging in my console has been really helpful alongside some other logging I'm doing.

**Review by watt — APPROVED** [review]

File: BlueprintUI/Sources/Blueprint View/BlueprintView.swift (hunk 1 above)
Comment (kyleve) [reply BlueprintUI/Sources/Blueprint View/BlueprintView.swift]: This is done

File: BlueprintUICommonControls/Sources/AccessibilityBlocker.swift (hunk 2 above)
Comment (kyleve) [reply BlueprintUICommonControls/Sources/AccessibilityBlocker.swift:22]: Yeah i dont think this works, gotta revert

Comment (kyleve) [reply BlueprintUICommonControls/Sources/AccessibilityBlocker.swift]: Reverted – would be good to fix this, but it can be a follow up.

**Pull request merged**